sklearn/utils/_metadata_requests.py

```diff
@@ -1494,7 +1494,12 @@ def _get_class_level_metadata_request_values(cls, method: str):
         # their parents.
         substr = f"__metadata_request__{method}"
         for base_class in reversed(inspect.getmro(cls)):
-            for attr, value in vars(base_class).items():
+            # Copy is needed with free-threaded context to avoid
+            # RuntimeError: dictionary changed size during iteration.
+            # copy.deepcopy applied on an instance of base_class adds
+            # __slotnames__ attribute to base_class.
```

Comment:
Could you extend a little bit this comment. Where does the deepcopy come from ?

Reply:
the traceback made it clearer. Not sure the comment alone would be enough for me to understand what's going on but I don't have a better suggestion :)